File: 조준장/level2/마법의 엘리베이터.py
def solution(storey):
    answer = 0
    
    while storey:
        current = storey % 10
        
        if current > 5:
            answer += 10 - current
            storey += 10
            
        elif current < 5:
            answer += current
            
        else:
            if (storey // 10) % 10 > 4:
                storey += 10
            answer += current
        
        storey //= 10
        
    return answer

File: 조준장/level2/마법의 엘리베이터.kt
class Solution {
    fun solution(storey: Int): Int {
        var answer: Int = 0
        
        var newStorey = storey
        
        while(newStorey != 0) {
            val n = newStorey % 10
            
            if (n > 5) {
                answer += 10 - n
                newStorey += n
            } else if (n < 5) {
                answer += n
                newStorey -= n
            } else {
                println((newStorey / 10) % 10)
                if ((newStorey / 10) % 10 > 4) {
                    newStorey += 10
                }
                answer += n
                
            }
           newStorey /= 10
        }
            
        return answer
    }
}
